dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/schema/DubboBeanDefinitionParser.java

```diff
@@ -96,9 +96,15 @@ private static RootBeanDefinition parse(Element element, ParserContext parserCon
                 generatedBeanName = beanClass.getName();
             }
             id = generatedBeanName;
-            int counter = 2;
-            while (parserContext.getRegistry().containsBeanDefinition(id)) {
-                id = generatedBeanName + (counter++);
+            if (parserContext.getRegistry().containsBeanDefinition(id)) {
+                if (beanClass.equals(ServiceBean.class)) {
+                    return (RootBeanDefinition) parserContext.getRegistry().getBeanDefinition(id);
```

Comment:
According to issue #7478, ignoring the duplicate service bean is not a good solution. Don't avoid the error. The problem of duplicate definition should be prompted to the user to deal with.

Reply:
should an exception be thrown?

Reply:
well, i fixed the code, can you review this pr again?